Polyclinic/Polyclinic.Application/Services/DoctorService.cs

```diff
@@ -0,0 +1,98 @@
+using Polyclinic.Application.Interfaces;
+using Polyclinic.Contracts.Dto;
+using Polyclinic.Domain.Interfaces;
+using Polyclinic.Domain.Subjects;
+
+namespace Polyclinic.Application.Services;
+
+/// <summary>
+/// Service implementation for Doctor business operations
+/// </summary>
+public class DoctorService(
+    IRepository<Doctor, int> doctorRepository,
+    IRepository<Specialization, int> specializationRepository
+    ) : IDoctorService
+{
+    public List<DoctorDto> GetAllDoctors()
+    {
+        var doctors = doctorRepository.ReadAll();
+        return [..doctors.Select(MapToDto)];
+    }
+
+    public DoctorDto? GetDoctorById(int id)
+    {
+        var doctor = doctorRepository.Read(id);
+        return doctor == null ? null : MapToDto(doctor);
+    }
+
+    public DoctorDto CreateDoctor(CreateDoctorRequest createRequest)
+    {
+        var specialization = specializationRepository.Read(createRequest.SpecializationId);
+        if (specialization == null)
+            throw new ArgumentException($"Specialization with ID {createRequest.SpecializationId} not found");
+
+        var doctor = new Doctor
+        {
+            PassportNumber = createRequest.PassportNumber,
+            FullName = createRequest.FullName,
+            YearOfBirth = createRequest.YearOfBirth,
+            Specialization = specialization,
+            Experience = createRequest.Experience
+        };
+
+        var id = doctorRepository.Create(doctor);
```

Comment:
Если вам не нужен id можно написать так:
```suggestion
        _ = doctorRepository.Create(doctor);
```